src/main/scala/mulAddSubRecodedFloatN.scala

```diff
@@ -296,6 +296,14 @@ class mulAddSubRecodedFloatN(sigWidth: Int, expWidth: Int, speed: Boolean = fals
           Mux(isNaNOut       , UInt(7 << expWidth-3), UInt(0, expWidth) )
   val fractOut = fractY | Fill(sigWidth, isNaNOut || isSatOut)
 
-  io.out := Cat(signOut, expOut, fractOut)
-  io.exceptionFlags := Cat(invalid, Bool(false), overflow, underflow, inexact)
+	val minPatchBypass1 = roundingMode_min && ~notSpecial_addZeros && signOut && (expOut(expWidth-1, expWidth-3) === UInt(0)) && totalUnderflowY && ~isZeroY	
+	val maxPatchBypass = roundingMode_max && ~notSpecial_addZeros && ~signOut && (expOut(expWidth-1, expWidth-3) === UInt(0)) && totalUnderflowY && ~isZeroY
+	val minPatchBypass2 = roundingMode_min && isZeroY && (signProd || opSignC) && ~notNaN_isInfOut 
+	val expOutPatched = Mux(maxPatchBypass || minPatchBypass1, UInt(minExp), expOut)
+	val signOutPatched = signOut || minPatchBypass2	
+	val outPatched = Cat(signOutPatched, expOutPatched, fractOut)
```

Comment:
Would y'all provide the test vectors that previously did not pass and now
do?  The unit was assumed to be compliant already, and if it is indeed not,
there is a hole in TestFloat.

On Mon, May 4, 2015 at 6:59 PM, Stevo <notifications@github.com> wrote:

> In src/main/scala/mulAddSubRecodedFloatN.scala
> <https://github.com/ucb-bar/berkeley-hardfloat/pull/3#discussion_r29640062>
> :
>
> > @@ -296,6 +296,14 @@ class mulAddSubRecodedFloatN(sigWidth: Int, expWidth: Int, speed: Boolean = fals
> >            Mux(isNaNOut       , UInt(7 << expWidth-3), UInt(0, expWidth) )
> >    val fractOut = fractY | Fill(sigWidth, isNaNOut || isSatOut)
> >
> > -  io.out := Cat(signOut, expOut, fractOut)
> > -  io.exceptionFlags := Cat(invalid, Bool(false), overflow, underflow, inexact)
> > +	val minPatchBypass1 = roundingMode_min && ~notSpecial_addZeros && signOut && (expOut(expWidth-1, expWidth-3) === UInt(0)) && totalUnderflowY && ~isZeroY	
> > +	val maxPatchBypass = roundingMode_max && ~notSpecial_addZeros && ~signOut && (expOut(expWidth-1, expWidth-3) === UInt(0)) && totalUnderflowY && ~isZeroY
> > +	val minPatchBypass2 = roundingMode_min && isZeroY && (signProd || opSignC) && ~notNaN_isInfOut
> > +	val expOutPatched = Mux(maxPatchBypass || minPatchBypass1, UInt(minExp), expOut)
> > +	val signOutPatched = signOut || minPatchBypass2	
> > +	val outPatched = Cat(signOutPatched, expOutPatched, fractOut)
>
> Sorry, we merged several changes. This code makes it IEEE compliant. The
> timing improvements are solely for the integer FMA (here:
> divSqrtRecodedFloat64.scala)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucb-bar/berkeley-hardfloat/pull/3/files#r29640062>.
>